src/manifest_store.rs

```diff
@@ -200,6 +203,33 @@ impl ManifestStore {
         Ok(())
     }
 
+    // TODO add tests
+    /// Delete a manifest from the object store.
+    pub(crate) async fn delete_manifest(&self, id: u64) -> Result<(), SlateDBError> {
+        // TODO add safety checks to prevent deletions of active manifests
+        let manifest_path = &self.get_manifest_path(id);
+        self.object_store.delete(manifest_path).await?;
+        Ok(())
+    }
+
+    pub(crate) async fn collect_garbage(&self, min_age: Duration) -> Result<(), SlateDBError> {
+        let manifest_path = &Path::from("/");
+        let mut files_stream = self.object_store.list(Some(manifest_path));
+        let min_age = chrono::Duration::from_std(min_age).expect("invalid duration");
+
+        // TODO Exclude snapshot manifests from GC when they are referenced by the current manifest
+        while let Some(file) = match files_stream.next().await.transpose() {
+            Ok(file) => file,
+            Err(e) => return Err(SlateDBError::ObjectStoreError(e)),
+        } {
+            if file.last_modified.signed_duration_since(Utc::now()) > min_age {
+                let id = self.parse_id(&file.location, "manifest")?;
+                self.delete_manifest(id).await?;
+            }
+        }
```

Comment:
I find nesting a `match` within a `while let` hard to reason about, what about:

```rust
loop {
    match file_stream.next().await {
        Ok(Some(file)) => {
            if file.last_modified.signed_duration_since(Utc::now()) > min_age {
                let id = self.parse_id(&file.location, "manifest")?;
                self.delete_manifest(id).await?;
            }
        },
        Ok(None) => return Ok(()), // or break, if you prefer
        Err(e) => return Err(SlateDBError::ObjectStoreError(e)),
    }
}
```

Reply:
Ends up looking like this because of the `transpose()`:

```rust
        loop {
            match files_stream.next().await {
                Some(Ok(file)) => {
                    if file.last_modified.signed_duration_since(Utc::now()) > min_age {
                        let id = self.parse_id(&file.location, "manifest")?;
                        self.delete_manifest(id).await?;
                    }
                    return Ok(());
                },
                None => return Ok(()),
                Some(Err(e)) => return Err(SlateDBError::ObjectStoreError(e)),
            }
        }
```

Reply:
Think I'd need to update `read_latest_manifest` to keep it consistent. Just making a note.